homeassistant/components/modbus/__init__.py

```diff
@@ -178,73 +146,136 @@ def write_coil(service):
         client_name = service.data[ATTR_HUB]
         hub_collect[client_name].write_coil(unit, address, state)
```

Comment:
All the writes are going to have to be async as well.

```
https://pymodbus.readthedocs.io/en/latest/source/example/async_asyncio_client.html
```

Reply:
write_coil is going to need the same

Reply:
have made all write async. I still have to test if I need to change hass.services.register call

Reply:
If you are calling it from async use
`hass.services.async_register`

Reply:
Thanks a lot, that call should be added to the async 101 page. I googled but did not find it.